test/e2e/oci/oci_test.go

```diff
@@ -0,0 +1,88 @@
+package test
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/open-policy-agent/opa/logging"
+	test_sdk "github.com/open-policy-agent/opa/sdk/test"
+	"github.com/open-policy-agent/opa/test/e2e"
+)
+
+func TestEnablePrintStatementsForBundles(t *testing.T) {
+	ref := "registry.io/someorg/somerepo:tag"
+	server := test_sdk.MustNewServer(test_sdk.MockOCIBundle(ref, map[string]string{
+		"post.rego": `
+		package peoplefinder.POST.api.users
+
+		import input.user.properties as user_props
+		
+		default allowed = false
+		
+		allowed {
+			user_props.department == "Operations"
+			user_props.title == "IT Manager"
+		}	
+		`,
+	}))
+	params := e2e.NewAPIServerTestParams()
+
+	buf := bytes.NewBuffer(nil)
+
+	logger := logging.New()
+	logger.SetOutput(buf)
+	params.Logger = logger
+
+	params.ConfigOverrides = []string{
+		"services.test.url=" + server.URL(),
+		"services.test.type=oci",
+		"bundles.test.resource=" + ref,
+	}
+
+	// Test runtime uses the local OCI image layers stored in download testdata that contain the
+	// rego policies based on the https://github.com/aserto-dev/policy-peoplefinder-abac template
+	e2e.WithRuntime(t, e2e.TestRuntimeOpts{WaitForBundles: true}, params, func(rt *e2e.TestRuntime) {
+
+		var readBuf []byte
+		type Props struct {
+			Department string `json:"department"`
+			Title      string `json:"title"`
+		}
+		type Attributes struct {
+			Properties Props `json:"properties"`
+		}
+		type Input struct {
+			User Attributes `json:"user"`
+		}
+
+		inputAllowed := Input{User: Attributes{Properties: Props{Department: "Operations", Title: "IT Manager"}}}
+
+		inputNotAllowed := Input{User: Attributes{Properties: Props{Department: "IT", Title: "Engineer"}}}
+
+		readBuf, err := rt.GetDataWithInput("peoplefinder/POST/api/users/allowed", inputAllowed)
+		if err != nil {
+			t.Fatal("failed to get data from runtime")
+		}
+
+		response := string(readBuf)
+		if !strings.Contains(response, "true") {
+			t.Fatalf("expected true but got: %s", response)
+		}
+
+		readBuf, err = rt.GetDataWithInput("peoplefinder/POST/api/users/allowed", inputNotAllowed)
+		if err != nil {
+			t.Fatal("failed to get data from runtime")
+		}
+
+		if !strings.Contains(string(readBuf), "false") {
+			t.Fatalf("expected true but got: %s", response)
+		}
+
+		expContains := "Bundle loaded and activated successfully"
```

Comment:
Hmm perhaps we should also check that the second pull did *not* do this? 🤔 Maybe I'm missing something... 🔍

Reply:
Rebased and added a verification that the bundle is skipped if it's the same on the pooling mechanism. I had to set the log level to debug in the test as the skip message is shown only on debug mode.